wezterm-gui/src/overlay/copy.rs

```diff
@@ -1141,13 +1155,37 @@ impl Pane for CopyOverlay {
                 (KeyCode::Char(c), KeyModifiers::NONE)
                 | (KeyCode::Char(c), KeyModifiers::SHIFT) => {
                     // Type to add to the pattern
-                    render.pattern.push(c);
+                    if render.pattern.capacity() - render.pattern.len() == 0 {
+                        render.pattern.reserve(10);
+                    }
```

Comment:
I don't think this reservation serves any useful purpose, and it just makes the code longer and more complex.

```suggestion
```

Reply:
Fixed